examples/ex1.c

```diff
@@ -0,0 +1,345 @@
+
+  // Apply a transformation to the mesh.
+  CeedScalar exact_vol = TransformMeshCoords(dim, mesh_size, mesh_coords);
+
+  // Context data to be passed to the 'f_build_mass' Q-function.
+  struct BuildContext build_ctx;
+  build_ctx.dim = build_ctx.space_dim = dim;
+
+  // Create the Q-function that builds the mass operator (i.e. computes its
+  // quadrature data) and set its context data.
+  CeedQFunction build_qfunc;
+  CeedQFunctionCreateInterior(ceed, 1, 1, sizeof(CeedScalar),
+                              (CeedEvalMode)(CEED_EVAL_GRAD|CEED_EVAL_WEIGHT),
+                              CEED_EVAL_NONE, f_build_mass,
+                              __FILE__":f_build_mass", &build_qfunc);
+  CeedQFunctionSetContext(build_qfunc, &build_ctx, sizeof(build_ctx));
+
+  // Create the operator that builds the quadrature data for the mass operator.
+  CeedOperator build_oper;
+  CeedOperatorCreate(ceed, mesh_restr, mesh_basis, build_qfunc, NULL, NULL,
+                     &build_oper);
+
+  // Compute the quadrature data for the mass operator.
+  CeedVector qdata;
+  CeedOperatorGetQData(build_oper, &qdata);
+  printf("Computing the quadrature data for the mass operator ...");
+  fflush(stdout);
+  CeedOperatorApply(build_oper, qdata, mesh_coords, NULL,
+                    CEED_REQUEST_IMMEDIATE);
+  printf(" done.\n");
+
+  // Create the Q-function that defines the action of the mass operator.
+  CeedQFunction apply_qfunc;
+  CeedQFunctionCreateInterior(ceed, 1, 1, sizeof(CeedScalar),
+                              CEED_EVAL_INTERP, CEED_EVAL_INTERP, f_apply_mass,
+                              __FILE__":f_apply_mass", &apply_qfunc);
+
+  // Create the mass operator.
+  CeedOperator oper;
+  CeedOperatorCreate(ceed, sol_restr, sol_basis, apply_qfunc, NULL, NULL,
+                     &oper);
+
+  // Compute the mesh volume using the mass operator: vol = 1^T.M.1.
+  printf("Computing the mesh volume using the formula: vol = 1^T.M.1 ...");
+  fflush(stdout);
+
+  // Create auxiliary solution-size vectors.
+  CeedVector u, v;
+  CeedVectorCreate(ceed, sol_size, &u);
+  CeedVectorCreate(ceed, sol_size, &v);
+
+  // Initialize 'u' with ones.
+  CeedScalar *u_host;
+  CeedVectorGetArray(u, CEED_MEM_HOST, &u_host);
+  for (CeedInt i = 0; i < sol_size; i++) {
+    u_host[i] = 1.;
+  }
+  CeedVectorRestoreArray(u, &u_host);
+
+  // Apply the mass operator: 'u' -> 'v'.
+  CeedOperatorApply(oper, qdata, u, v, CEED_REQUEST_IMMEDIATE);
+
+  // Compute and print the sum of the entries of 'v' giving the mesh volume.
+  const CeedScalar *v_host;
+  CeedVectorGetArrayRead(v, CEED_MEM_HOST, &v_host);
+  CeedScalar vol = 0.;
+  for (CeedInt i = 0; i < sol_size; i++) {
+    vol += v_host[i];
+  }
+  CeedVectorRestoreArrayRead(v, &v_host);
+  printf(" done.\n");
+  printf("Exact mesh volume    : % .14g\n", exact_vol);
+  printf("Computed mesh volume : % .14g\n", vol);
+  printf("Error                : % .14g\n", vol-exact_vol);
```

Comment:
Print the absolute value?

Reply:
I left the sign on purpose. Do you insist on absolute value?

Reply:
No